### PR TITLE
chore(gatsby): Migrate from `source-map` to `@jridgewell/trace-mapping`

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Due to https://github.com/nodejs/node/issues/42638 the `source-map` package is breaking in Node v18. We can migrate to https://github.com/jridgewell/trace-mapping like Jest (https://github.com/facebook/jest/pull/12692) and Istanbul (https://github.com/istanbuljs/v8-to-istanbul/pull/186) did.

TODOs:

- [x] Get https://github.com/jridgewell/trace-mapping/issues/7 working (thus this PR is in draft)
- [x] All tests pass (including on Node 18, [separate PR](https://github.com/gatsbyjs/gatsby/pull/35622)). Tests can pass once https://github.com/jridgewell/trace-mapping/issues/9 is fixed
- [x] Manual testing with stack traces in error overlay in `gatsby develop` while using Node 18

### Documentation

No update needed.

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/35607
[ch50439]

### CODE REVIEW

File: packages/gatsby-cli/src/reporter/prepare-stack-trace.ts [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
-import {
-  SourceMapConsumer,
-  BasicSourceMapConsumer,
-  IndexedSourceMapConsumer,
-  NullableMappedPosition,
-} from "source-map"
+import { TraceMap, originalPositionFor } from "@jridgewell/trace-mapping"
 import * as path from "path"
 
+interface INullableMappedPosition {
```
Comment (jridgewell) [inline packages/gatsby-cli/src/reporter/prepare-stack-trace.ts:11]: Nit: We provide `OriginalMapping` and `InvalidOriginalMapping` types that can be used for discriminated unions.

Comment (LekoArts) [reply packages/gatsby-cli/src/reporter/prepare-stack-trace.ts:11]: Good to know 👍

File: packages/gatsby/src/utils/stack-trace-utils.ts [hunk 2]
```diff
@@ -111,33 +111,36 @@ export const getNonGatsbyCodeFrameFormatted = ({
...
     return {
-      sourcePosition,
-      sourceContent,
+      sourcePosition: null,
+      sourceContent: null,
     }
-  })
+  }
+
+  console.log({ sourcesContent })
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply packages/gatsby/src/utils/stack-trace-utils.ts:136]: I guess for now, yes 😅 I wanted to check if it contains the information I need but was distracted by other things. Just want to figure out how to migrate `sourceContentFor` :)

**Comment by LekoArts** [comment]
I still need to do some manual testing with this

**Pull request merged**